src/compiler/program.ts

```diff
@@ -4593,10 +4594,10 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
                 }
                 else {

```

Comment:
Would probably be even better to add new diags in the middle here that says "is deprecated and has stopped functioning", since it'll be in future tense even when we're on like TS 5.6.